File: data/snippets/quiz011.cpp
class Order {
private:
    int orderId;
    std::string customerName;
    double totalAmount;
    std::vector<std::string> items;
    std::string shippingAddress;
    std::string paymentMethod;
    
public:
    Order(int id, std::string name) : orderId(id), customerName(name), totalAmount(0) {}
    
    // 因为支付系统变化需要修改
    void processPayment() {
        std::cout << "Processing payment for order #" << orderId << std::endl;
        
        if (paymentMethod == "CreditCard") {
            std::cout << "Processing credit card payment..." << std::endl;
            // 信用卡支付逻辑
            verifyCardDetails();
            processCreditCardTransaction();
        } 
        else if (paymentMethod == "PayPal") {
            std::cout << "Processing PayPal payment..." << std::endl;
            // PayPal支付逻辑
            initiatePayPalAuthentication();
            finalizePayPalTransaction();
        }
        else if (paymentMethod == "BankTransfer") {
            std::cout << "Processing bank transfer..." << std::endl;
            // 银行转账逻辑
            generateBankTransferInstructions();
            waitForTransferConfirmation();
        }
    }
    
    // 因为物流系统变化需要修改
    void arrangeShipping() {
        std::cout << "Arranging shipping for order #" << orderId << std::endl;
        
        // 跟踪信息生成
        std::string trackingNumber = generateTrackingNumber();
        
        // 货运公司选择逻辑
        std::string carrier = selectBestCarrier();
        
        // 标签打印逻辑
        printShippingLabel(carrier, trackingNumber);
        
        // 包装说明
        generatePackagingInstructions();
    }
    
    // 因为库存系统变化需要修改
    void updateInventory() {
        std::cout << "Updating inventory for order #" << orderId << std::endl;
        
        // 库存扣减逻辑
        for (const auto& item : items) {
            reduceStockLevel(item);
        }
        
        // 库存预警检查
        checkLowStockLevels();
        
        // 自动补货逻辑
        triggerAutomaticReordering();
    }
    
    // 因为报告系统变化需要修改
    void generateReports() {
        std::cout << "Generating reports for order #" << orderId << std::endl;
        
        // 销售报告
        createSalesReport();
        
        // 税务报告
        generateTaxDocuments();
        
        // 客户消费报告
        updateCustomerSpendingHistory();
    }
    
    // 各种辅助方法
    void verifyCardDetails() { /* 信用卡验证逻辑 */ }
    void processCreditCardTransaction() { /* 信用卡交易处理 */ }
    void initiatePayPalAuthentication() { /* PayPal认证 */ }
    void finalizePayPalTransaction() { /* PayPal交易完成 */ }
    void generateBankTransferInstructions() { /* 生成银行转账说明 */ }
    void waitForTransferConfirmation() { /* 等待转账确认 */ }
    std::string generateTrackingNumber() { return "TRK12345"; }
    std::string selectBestCarrier() { return "FastShip"; }
    void printShippingLabel(std::string carrier, std::string tracking) { /* 打印标签 */ }
    void generatePackagingInstructions() { /* 生成包装说明 */ }
    void reduceStockLevel(std::string item) { /* 减少库存 */ }
    void checkLowStockLevels() { /* 检查低库存 */ }
    void triggerAutomaticReordering() { /* 触发自动补货 */ }
    void createSalesReport() { /* 创建销售报告 */ }
    void generateTaxDocuments() { /* 生成税务文档 */ }
    void updateCustomerSpendingHistory() { /* 更新客户消费历史 */ }
};

File: data/snippets/quiz011.java
import java.util.List;
import java.util.ArrayList;

public class Order {
    private int orderId;
    private String customerName;
    private double totalAmount;
    private List<String> items;
    private String paymentMethod;
    
    public Order(int id, String name) {
        this.orderId = id;
        this.customerName = name;
        this.totalAmount = 0;
        this.items = new ArrayList<>();
    }
    
    // Needs modification when payment systems change
    public void processPayment() {
        System.out.println("Processing payment for order #" + orderId);
        
        if ("CreditCard".equals(paymentMethod)) {
            System.out.println("Processing credit card payment...");
            verifyCardDetails();
            processCreditCardTransaction();
        } else if ("PayPal".equals(paymentMethod)) {
            System.out.println("Processing PayPal payment...");
            initiatePayPalAuthentication();
            finalizePayPalTransaction();
        } else if ("BankTransfer".equals(paymentMethod)) {
            System.out.println("Processing bank transfer...");
            generateBankTransferInstructions();
            verifyBankTransfer();
        }
    }
    
    // Needs modification when shipping systems change
    public void arrangeShipping() {
        System.out.println("Arranging shipping for order #" + orderId);
        
        String trackingNumber = generateTrackingNumber();
        String carrier = selectBestCarrier();
        printShippingLabel(carrier, trackingNumber);
        calculateShippingCosts();
        schedulePickup();
    }
    
    // Needs modification when inventory systems change
    public void updateInventory() {
        System.out.println("Updating inventory for order #" + orderId);
        
        for (String item : items) {
            reduceStockLevel(item);
            updateProductAvailability(item);
        }
        
        checkLowStockLevels();
        triggerAutomaticReordering();
    }
    
    // Needs modification when reporting systems change
    public void generateReports() {
        System.out.println("Generating reports for order #" + orderId);
        
        createSalesReport();
        generateTaxDocuments();
        updateCustomerSpendingHistory();
    }
    
    // Helper methods
    private void verifyCardDetails() { /* Credit card verification logic */ }
    private void processCreditCardTransaction() { /* Credit card transaction processing */ }
    private void initiatePayPalAuthentication() { /* PayPal authentication */ }
    private void finalizePayPalTransaction() { /* PayPal transaction completion */ }
    private void generateBankTransferInstructions() { /* Generate bank transfer instructions */ }
    private void verifyBankTransfer() { /* Verify bank transfer */ }
    
    private String generateTrackingNumber() { return "TRK12345"; }
    private String selectBestCarrier() { return "FastShip"; }
    private void printShippingLabel(String carrier, String tracking) { /* Print label */ }
    private void calculateShippingCosts() { /* Calculate shipping costs */ }
    private void schedulePickup() { /* Schedule pickup */ }
    
    private void reduceStockLevel(String item) { /* Reduce inventory */ }
    private void updateProductAvailability(String item) { /* Update availability */ }
    private void checkLowStockLevels() { /* Check low inventory */ }
    private void triggerAutomaticReordering() { /* Trigger automatic reordering */ }
    
    private void createSalesReport() { /* Create sales report */ }
    private void generateTaxDocuments() { /* Generate tax documents */ }
    private void updateCustomerSpendingHistory() { /* Update customer spending history */ }
}
